components/switch/switch-mixin.js

```diff
@@ -208,7 +214,7 @@ export const SwitchMixin = superclass => class extends FocusMixin(RtlMixin(Focus
 	}
 
 	get _labelContent() {
-		return html`${this.text}`;
+		return html`<span @click='${this._handleClick}' @mouseover='${this._handleSwitchHover}' @mouseout='${this._handleSwitchHoverLeave}'>${this.text}</span>`;
```

Comment:
Could the hover effects not have just been applied whenever the entire host element is hovered, vs. listening for mouse over/out for both the inner container and the label?

Reply:
So if we were to do that, i.e.
```css
:host(:hover) .d2l-switch-inner {
				border-color: var(--d2l-color-celestine);
				box-shadow: 0 0 0 1px var(--d2l-color-celestine) inset;
			}
```
then the hover styling would also apply to the entire label, including the `conditions met` part of the label in the visibility switch. 

However, I got [confirmation from Jeff](https://d2l.slack.com/archives/C03FC0GDVB7/p1658932976745289?thread_ts=1658519924.626809&cid=C03FC0GDVB7) that we don't want that to happen.

Reply:
Gotcha. I think you can still remove the mouseover/mouseout on the `d2l-switch-inner` and modify its CSS a bit to handle its own hover state:

```css
.d2l-switch-inner:hover,
.switch-hover {
	border-color: var(--d2l-color-celestine);
	box-shadow: 0 0 0 1px var(--d2l-color-celestine) inset;
}
```

That way you only have the one set of event listeners.